### DISCUSSION TITLE
Unable to set mysql_ssl and mysql_ssl_two_way to 'true

### DISCUSSION THREAD

**Discussion by noopur100** [TPC-Council/HammerDB#434] [Q&A] [answered]
Hi Team,

I am trying to mysql_ssl and mysql_ssl_two_way to 'true' which is by default set to 'false', but when I try it using below command 
 I get error as below :
hammerdb>diset tpcc mysql_ssl true
Usage: diset dict key value
Dictionary "tpcc" for MySQL exists but key "mysql_ssl" doesn't
Type "print dict" for valid dictionaries and keys for MySQL

#diset tpcc mysql_ssl true
#diset tpcc mysql_ssl_two_way true

I tried to manually update the values in file mysql.xml (/hammerdb/HammerDB-4.5/config/mysql.xml), buts still when I print dict the mysql_ssl and mysql_ssl_two_way are set to false.
I need to set ssl to true, as without setting ssl value to true I am unable to connect hammerdb to mysql instance to load tpcc data.

Below is my print dict output:

hammerdb>print dict
Dictionary Settings for MySQL
connection {
 mysql_host               = 35.239.232.40
 mysql_port               = 3306
 mysql_socket             = null
 mysql_ssl                = true
 mysql_ssl_two_way        = false
 mysql_ssl_linux_capath   = /etc/mysql/certs/
 mysql_ssl_windows_capath = C:\mysql\certs
 mysql_ssl_ca             = ca-cert.pem
 mysql_ssl_cert           = client-cert.pem
 mysql_ssl_key            = client-key.pem
 mysql_ssl_cipher         = server
}
tpcc       {
 mysql_count_ware       = 2
 mysql_num_vu           = 2
 mysql_user             = root
 mysql_pass             = password321
 mysql_dbase            = tpcc
 mysql_storage_engine   = innodb
 mysql_partition        = true
 mysql_prepared         = false
 mysql_total_iterations = 10000000
 mysql_raiseerror       = false
 mysql_keyandthink      = false
 mysql_driver           = test
 mysql_rampup           = 2
 mysql_duration         = 5
 mysql_allwarehouse     = false
 mysql_timeprofile      = false
 mysql_async_scale      = false
 mysql_async_client     = 10
 mysql_async_verbose    = false
 mysql_async_delay      = 1000
 mysql_connect_pool     = false
}

root@dstream-poc:/etc/mysql/certs# ls -ltr
total 12
-rwxrwxrwx 1 root root 1268 Sep 22 08:51 client-cert.pem
-rwxrwxrwx 1 root root 1272 Sep 22 08:52 ca-cert.pem
-rwxrwxrwx 1 root root 1674 Sep 22 08:52 client-key.pem

**Comment by sm-shaw** [comment, marked as answer]
Hi, It's in the connection sub-dict so use connection rather than tpcc. 

```
HammerDB CLI v4.5
Copyright (C) 2003-2022 Steve Shaw
Type "help" for a list of commands
hammerdb>dbset db mysql
Database set to MySQL

hammerdb>diset connection mysql_ssl true
Changed connection:mysql_ssl from false to true for MySQL

hammerdb>diset connection mysql_ssl_two_way true
Changed connection:mysql_ssl_two_way from false to true for MySQL

hammerdb>print dict
Dictionary Settings for MySQL
connection {
 mysql_host               = localhost
 mysql_port               = 3306
 mysql_socket             = null
 mysql_ssl                = true
 mysql_ssl_two_way        = true
 mysql_ssl_linux_capath   = /etc/mysql/certs
 mysql_ssl_windows_capath = C:\mysql\certs
 mysql_ssl_ca             = ca-cert.pem
 mysql_ssl_cert           = client-cert.pem
 mysql_ssl_key            = client-key.pem
 mysql_ssl_cipher         = server
}
tpcc       {
 mysql_count_ware       = 4
 mysql_num_vu           = 4
 mysql_user             = root
 mysql_pass             = mysql
 mysql_dbase            = tpcc
 mysql_storage_engine   = innodb
 mysql_partition        = false
 mysql_prepared         = false
 mysql_total_iterations = 10000000
 mysql_raiseerror       = false
 mysql_keyandthink      = false
 mysql_driver           = timed
 mysql_rampup           = 2
 mysql_duration         = 5
 mysql_allwarehouse     = false
 mysql_timeprofile      = false
 mysql_async_scale      = false
 mysql_async_client     = 10
 mysql_async_verbose    = false
 mysql_async_delay      = 1000
 mysql_connect_pool     = false
}

hammerdb>
```

**Reply by noopur100** [reply]
Steve thank you so much, you are the Saviour... provided command solved my issue :)

**Reply by sm-shaw** [reply]
No problem, this gives the idea that the error message could potentially be enhanced to say something like the following to make this type of error clearer, 

`Dictionary "tpcc" for MySQL exists but key "mysql_ssl" doesn't, key "mysql_ssl" is in the "connection" dictionary.`

`Dictionary "tpcc" for MySQL exists but key "mysql_sst" doesn't, key "mysql_sst" cannot be found in any MySQL dictionary.`

as it is straightforward to search for keys.

**Reply by sm-shaw** [reply]
PR #437 adds this change so error message now reports if the incorrect dictionary is chosen to make it clearer.

**Reply by noopur100** [reply]
Great..!! It will now be crystal clear if someone faces issue again...great work
